sdk/src/account.rs

```diff
@@ -812,8 +812,9 @@ pub fn is_executable(account: &impl ReadableAccount, feature_set: &FeatureSet) -
     if !feature_set.is_active(&deprecate_executable_meta_update_in_bpf_loader::id()) {
         account.executable()
     } else {
-        // First, check if the account is empty. Empty accounts are not executable.
-        if account.data().is_empty() {
+        // First, check if the account is empty or all zeros. Empty or all zeros
+        // accounts are not executable.
+        if account.data().is_empty() || account.data().iter().all(|&x| x == 0) {
```

Comment:
That check is too expensive, better to test if the ELF header exists in RBPF.

Reply:
yeah. That's sound like a better idea.